### PR TITLE
Project 5 - Therese Hagelin

### PR DESCRIPTION
**Pull request by ThereseHag**
I have now created a Guess Who game with help from Javascript. 
I used the characters already provided, but added some more options to make it possible to narrow down even more. 

I struggled to get the filterCharacters to work, but finally made it after defining both category and value. I also added event listeners in order to make both restart button and play again button work. 

I have made small changes to the CSS file in order to get colors of my own choice, but no further changes there.

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -240,104 +420,166 @@ const generateBoard = () => {
...
+  const {category, attribute, value} = currentQuestion
+  if (category === "accessories") {
     if (keep) {
       alert(
         `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[attribute] === value
+      );
```
Comment (TinaJuren) [inline code/script.js:489]: I know that there are ways to shorten the code by writing the filtering for all the characters. But, as a beginner I find it very easy to understand the flow of the game with your way of writing it. When writing my own code I ended up confused and in a way I wish I did it your way instead :)

**Review by TinaJuren — APPROVED** [review]
I'm very impressed by your project, Therese! Your code is clean and well structured. I also find it easy to follow along since you added informative comments throughout the code.

File: code/script.js [hunk 2]
```diff
@@ -1,235 +1,415 @@
...
+    img: "images/jabala.svg",
+    hairColor: "hidden",
+    eyeColor: "hidden",
     glasses: true,
     hat: true,
+    patch: false,
+    earrings: false,
+    cell: false,
+    necklace: false,
+    parrot: false,
```
Comment (KarolCislowski) [inline code/script.js:23]: It is great that you experimented with data provided by teachers. 🥇

File: code/script.js [hunk 3]
```diff
@@ -240,104 +420,166 @@ const generateBoard = () => {
...
+  const keep = currentQuestion.value === secret[currentQuestion.attribute];
+  filterCharacters(keep);
+};
 
-// It'll filter the characters array and redraw the game board.
+// Triggers an alert telling whether the secret person has the attribute chosen or not, and redraws a filtered board.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
+  const {category, attribute, value} = currentQuestion
```
Comment (KarolCislowski) [inline code/script.js:481]: Great deconstruction!

**Review by KarolCislowski — APPROVED** [review]
This is an awesome nice project 🌈 

Remember that you can always come back to it when you have some spare time and add more black level functionalities to make it an even greater portfolio asset! 🦄

**Pull request closed without merging**